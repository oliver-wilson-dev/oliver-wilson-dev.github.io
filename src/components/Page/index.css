@import "../../styles/shared.css";

.page {
    max-width: 100%;
    margin: 0;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    flex-grow: 1;
}

@media (min-width: 1024px) {
    .page {
        margin: 20px auto;
    }
}
