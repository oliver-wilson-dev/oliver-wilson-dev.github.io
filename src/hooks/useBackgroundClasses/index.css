.themeBackground {
  background-color: var(--primaryBackgroundLight);
  transition: background-color 0.4s ease-in;

  &__dark {
    background-color: var(--primaryBackgroundDark);
  }
}
