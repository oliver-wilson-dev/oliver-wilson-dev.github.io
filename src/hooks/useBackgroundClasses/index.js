import { useMemo } from 'react';
import classnames from 'classnames';
import styles from './index.css';
import { DARK_THEME } from '../../state/actions/constants';

const useBackgroundClasses = ({ theme }) => useMemo(
  () => classnames(styles.themeBackground, {
    [styles.themeBackground__dark]: theme === DARK_THEME
  }),
  [theme]
);

export default useBackgroundClasses;
