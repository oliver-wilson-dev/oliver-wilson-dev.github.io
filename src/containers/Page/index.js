import { connect } from 'react-redux';
import Page from '../../components/Page';
import { getTheme } from '../../state/selectors';

const mapStateToProps = state => ({
  theme: getTheme(state)
});

export default connect(mapStateToProps, undefined)(Page);
