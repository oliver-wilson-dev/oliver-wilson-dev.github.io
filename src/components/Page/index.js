import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import useBackgroundClasses from '../../hooks/useBackgroundClasses';

import styles from './index.css';

const Page = ({ additionalStyles, children, theme }) => {
  const backgroundClasses = useBackgroundClasses({ theme });

  return (
    <div className={cn(styles.page, backgroundClasses, additionalStyles)}>{children}</div>
  );
};

Page.propTypes = {
  additionalStyles: PropTypes.string,
  theme: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default Page;
