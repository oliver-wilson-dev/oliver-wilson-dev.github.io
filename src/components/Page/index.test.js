import React from 'react';
import { shallow } from 'enzyme';
import Page from '.';
import useBackgroundClasses from '../../hooks/useBackgroundClasses';

jest.mock('../../hooks/useBackgroundClasses');

const render = (props = {}) => shallow(<Page {...props}><h1>hello world</h1></Page>);

describe('Page', () => {
  const mockBackgroundClasses = 'test-background-classes';
  beforeEach(() => {
    useBackgroundClasses.mockReturnValue(mockBackgroundClasses);
  });


  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when provided with an additionalStyles prop', () => {
    it('should apply that class to the wrapper div', () => {
      const mockClass = 'test-additional-styles';
      const component = render({ additionalStyles: mockClass });

      expect(component.find(`.${mockClass}`).exists()).toBe(true);
    });
  });
});
