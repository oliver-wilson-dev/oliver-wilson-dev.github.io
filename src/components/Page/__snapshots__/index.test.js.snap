// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render correctly 1`] = `
<div
  className="page test-background-classes"
>
  <h1>
    hello world
  </h1>
</div>
`;
